File: desktop/src/app/schedule-information/serviceChange.request.ts
/**
 * This class defines a model for requests to change service information in simulation mode in TraMS.
 */
export class ServiceChangeRequest {

    /**
     * Company which services should be reset for
     */
    private company: string;

    /**
     * The id of the service.
     */
    private serviceId: string;

    /**
     * The id of the route and schedule running the service e.g. 3/1
     */
    private scheduleId: string;

    /**
     * A boolean which is true iff the service is out of service i.e. not running.
     */
    private outOfService: boolean;

    /**
     * Allow a service to start after normal stop to reduce delays etc.
     */
    private tempStartStopPos: number;

    /**
     * Allow a service to end before normal stop to reduce delays etc.
     */
    private tempEndStopPos: number;

    /**
     * Construct a new request to change a service.
     * @param company the company that we should change a service for.
     * @param serviceId the id of the service to change / update.
     * @param scheduleId the id of the schedule to change in the format route and schedule number e.g. 3/1
     * @param outOfService a boolean which is true iff the service should not be in service.
     * @param tempStartStopPos the start position of this service when it has been shortened (default: 0).
     * @param tempEndStopPos the end position of this service when it has been shortened (default: 0 or stopList size -1).
     */
    constructor(company: string, serviceId: string, scheduleId: string, outOfService: boolean,
                tempStartStopPos: number, tempEndStopPos: number) {
        this.company = company;
        this.serviceId = serviceId;
        this.scheduleId = scheduleId;
        this.outOfService = outOfService;
        this.tempStartStopPos = tempStartStopPos;
        this.tempEndStopPos = tempEndStopPos;
    }
}

File: server/src/main/java/de/davelee/trams/server/request/ServiceChangeRequest.java
package de.davelee.trams.server.request;

/**
 * This class is part of the TraMS Server REST API. It represents a request to change the service information
 * because service should be shortened etc.
 * @author Dave Lee
 */
public class ServiceChangeRequest {

    /**
     * Company which services should be reset for
     */
    private String company;

    /**
     * The id of the service.
     */
    private String serviceId;

    /**
     * The id of the route and schedule running the service e.g. 3/1
     */
    private String scheduleId;

    /**
     * A boolean which is true iff the service is out of service i.e. not running.
     */
    private boolean outOfService;

    /**
     * Allow a service to start after normal stop to reduce delays etc.
     */
    private int tempStartStopPos;

    /**
     * Allow a service to end before normal stop to reduce delays etc.
     */
    private int tempEndStopPos;

    public ServiceChangeRequest() {
    }

    public ServiceChangeRequest(String company, String serviceId, String scheduleId, boolean outOfService, int tempStartStopPos, int tempEndStopPos) {
        this.company = company;
        this.serviceId = serviceId;
        this.scheduleId = scheduleId;
        this.outOfService = outOfService;
        this.tempStartStopPos = tempStartStopPos;
        this.tempEndStopPos = tempEndStopPos;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getServiceId() {
        return serviceId;
    }

    public void setServiceId(String serviceId) {
        this.serviceId = serviceId;
    }

    public String getScheduleId() {
        return scheduleId;
    }

    public void setScheduleId(String scheduleId) {
        this.scheduleId = scheduleId;
    }

    public boolean isOutOfService() {
        return outOfService;
    }

    public void setOutOfService(boolean outOfService) {
        this.outOfService = outOfService;
    }

    public int getTempStartStopPos() {
        return tempStartStopPos;
    }

    public void setTempStartStopPos(int tempStartStopPos) {
        this.tempStartStopPos = tempStartStopPos;
    }

    public int getTempEndStopPos() {
        return tempEndStopPos;
    }

    public void setTempEndStopPos(int tempEndStopPos) {
        this.tempEndStopPos = tempEndStopPos;
    }
}
